pkg/operator/watchdog.go

```diff
@@ -115,17 +129,36 @@ forloop:
 	return nil
 }
 
-func (dw *DaprWatchdog) listPods(ctx context.Context) {
+func (dw *DaprWatchdog) listPods(ctx context.Context) bool {
 	log.Infof("DaprWatchdog started checking pods")
 
+	// Look for the dapr-sidecar-injector deployment first and ensure it's running
+	// Otherwise, the pods would come back up without the sidecar again
+	deployment := &appsv1.DeploymentList{}
+	err := dw.client.List(ctx, deployment, &client.ListOptions{
```

Comment:
If we were to list only in the Dapr system namespace, would that result in removing the list deployments in all namespaces permission or do we have that anyway l?

Reply:
We still need to list the workload pods, correct. 

For the sidecar injector, I'm listing the deployments rather than the pods, to be able to check if the deployment has at least 1 pod in "ready" state.